### PR TITLE
[Blog] Implement New Design To Existing NetlifyCMS /Blog Page

### LINKED ISSUE #140: [Blog] Implement new design to existing netlifyCMS /blog page
market blog

## Gitcoin Bounty Hunters ##
Please only consider picking up this issue if you have extensive experience with react-static and feel confident that you are able to implement the designs with a high fidelity to the provided assets. Additionally, responsiveness of the implementation is very important and the pages and design must translate well across, mobile, tablet, and large monitor resolutions.

## Before you `start work`

Please read our contribution [guidelines](https://docs.marketprotocol.io/#contributing) and if there is a bounty involved  please also see [here](https://docs.marketprotocol.io/#gitcoin-and-bounties)

If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please make sure to comment in the issue here immediately after starting work so we know your plans for implementation and a timeline.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

### Why Is this Needed?

*Summary*: Brings our blog up to date with the rest of our design and pages.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Solution
[comment]: # (Provide a summary of the solution and a task list on what needs to be fixed.)
*Summary*: Using [.sketch file](https://github.com/MARKETProtocol/assets/blob/master/MockUps/MARKET_Protocol-blog.sketch) implement new design using react-static.

### Definition of Done

- [ ] Using the provided assets, integrate new design into the /blog page that uses netlifyCMS
- [ ] Update tests as needed 
- [ ] Add new tests for all newly generated code
- [ ] For reference the existing implementation exists here http://dev.website.marketprotocol.io/blog
- [ ] Interact with @MARKETProtocol/core and expect a few revisions

### Additional comments
- The category column should stay fixed once it reaches to the bottom scroll position

### PR DESCRIPTION
**Pull request by hbeckeri**
##### Description
Implement New Design To Existing NetlifyCMS /Blog Page

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
UI

##### Screenshots (Optional) 
localhost_3000_blog
jun-17-2018 12-51-25

##### Testing
I created tests for new ui components to ensure they render without any errors.

##### Refers/Fixes
Resolves #140

### CODE REVIEW

**Comment by hbeckeri** [comment]
I will continue to improve test coverage, but would appreciate visual and code reviews

**Comment by nitinrgupta** [comment]
@hbeckeri thanks for the PR! 
I briefly went through the changes and noticed that you haven't used `styled-components`. Maybe take advantage of it since we're using it across the project and keeping it consistent?

**Comment by eswarasai** [comment]
@hbeckeri -- Can you please squash all your commits and also rebase from `upstream/develop`? Please avoid merging `upstream:develop` into your branch. Thanks!

**Comment by pelsasser** [comment]
@hbeckeri  - thanks for the PR! I am just going through a visual review now.  Here are my notes

- No left margin
  - please see the .sketch file or screenshot in the original issue for the layout
- Twitter and Facebook links on the posts should allow people to share these posts not just link back to our social. (Please replace the medium logo on the posts with facebook).  They also appear a touch large compared to the mock.
- `Sample Post 4` appears twice. 
- Please remove the Your Email Here Box and wire up the button the same as the `Click To Subscribe` button that is found [here ](http://dev.website.marketprotocol.io/#subscribe)

@nitinrgupta can you please also review? - deployed [here](http://dev.website.marketprotocol.io/blog/) for easy visual reference

**Comment by hbeckeri** [comment]
@pelsasser Is there a certain format (ie. copy and language) for sharing the blog posts? Are sharing features already created in the project? If not, for twitter I can use [web-intents](https://developer.twitter.com/en/docs/twitter-for-websites/web-intents/overview.html) which are just urls. For facebook I see they have a [share-dialog](https://developers.facebook.com/docs/sharing/reference/share-dialog), but it requires some setup with a facebook app.

**Comment by eswarasai** [comment]
@hbeckeri -- Might need to handle the Blog UI for larger screens as well I guess. Please find the screenshot from the MacOS Google Chrome browser with the issue.

screen shot 2018-06-18 at 9 23 09 pm

**Comment by nitinrgupta** [comment]
@hbeckeri Apart from the responsiveness issues, I found the following:

- [ ] The card body text should be a `font-size: 14px` and `line-height: 20px`. Please refer the sketch files for the font sizes. 
screen shot 2018-06-18 at 9 46 21 pm

____________________________________________________________________________________________________________

- [ ] Some alignment issues over here.
screen shot 2018-06-18 at 9 43 18 pm

____________________________________________________________________________________________________________

- [ ] The right sidebar is not scrollable till the bottom. It cuts off at a certain scroll height. 
screen shot 2018-06-18 at 9 42 59 pm


____________________________________________________________________________________________________________

- [ ] The load more button can come at the end of blog cards instead at the end of the entire blog page on resolutions below `768px`

File: src/components/Blog/CategorySelector.tsx [hunk 1]
```diff
@@ -13,15 +43,22 @@ export default ({
...
-          key={`cat#${i}`}
-          cat={cat}
-          showBorder={selectedCat && selectedCat === cat}
-          onClick={() => (selectedCat === cat ? null : onSelectCat(cat))}
-        />
-      ))}
-    </div>
+    <RootWrap>
+      <div className="root">
+        <h2 className="title">About Our Blog</h2>
```
Comment (nitinrgupta) [inline src/components/Blog/CategorySelector.tsx]: @hbeckeri We haven't used conventional `className` stylings across the project, instead please tackle these using `styled-components`.

**Comment by hbeckeri** [comment]
@eswarasai @nitinrgupta Many of these issues I have already fixed. It seems like the deployed version does not auto deploy when I commit changes, so you'll need to review my PR from my branch.

@nitinrgupta in response to "The right sidebar is not scrollable till the bottom. It cuts off at a certain scroll height.", The issue says that the category section needs to be fixed when it reaches the bottom scroll position. The only way to prevent the cut off would be to reorder the sidebar sections by putting the category selector last.

**Comment by nitinrgupta** [comment]
@hbeckeri I get it. Let's not do it in that case if it affects the usability. But make sure the right sidebar should stay if it reaches the bottom of the sidebar height.

**Comment by hbeckeri** [comment]
@nitinrgupta Are you asking me to remove or keep the fixed treatment on the category selector?

**Comment by nitinrgupta** [comment]
@hbeckeri let's not give a fixed treatment for the categories section. But the sidebar should be fixed once it reaches to the bottom position just like how we had earlier.

**Comment by hbeckeri** [comment]
@nitinrgupta We will still see the clipping if I fix the entire sidebar at the bottom scroll position. I don't understand how it would be much different than fixing the category section

**Comment by eswarasai** [comment]
@hbeckeri -- One small issue is that if you scroll till the bottom of the page, on top of footer, you can still see a bit of social links box, just like in the screenshot below.

screen shot 2018-06-19 at 8 26 31 pm

**Comment by eswarasai** [comment]
@hbeckeri -- Looks like there's a UX issue when filtering the blogs via categories. Firstly there's no clear indicator that blog posts are being filtered and also on how to clear the selected category. Secondly, the category filter section is being hidden by the footer upon filtering by category.

I wonder if having the category selected as URL param would help with easy linking maybe. @pelsasser -- Any thoughts on this so that we can ask @hbeckeri to care of it in this PR?

**Comment by pelsasser** [comment]
@eswarasai - good points.  I think this is outside of the scope of this issue, ca you (@eswarasai ) create a new issues for the filtering changes and needs?

**Comment by pelsasser** [comment]
@hbeckeri - there are also several issues with the mobile version of the site.

**Comment by eswarasai** [comment]
@pelsasser -- Sure. I can create an issue for the categories filter UI/UX.

@hbeckeri -- One more thing I noticed was I couldn't find Recent Tweets section in the right sidebar. Did we remove it or is it not scrollable?

**Comment by hbeckeri** [comment]
@pelsasser what issues are you seeing on mobile?

**Comment by pelsasser** [comment]
@hbeckeri - sorry for not being more clear, I hopped on a call mid comment.  Main issue is around the mobile layout

For instance the "About our Blog" / "Categories" Section appears at the very end of the screen.

**Comment by pelsasser** [comment]
signing off with current implementation of the sidebar in order to move things along here (@nitinrgupta we can always iterate ;)

**Comment by nitinrgupta** [comment]
@hbeckeri changes look good now 👍 
I noticed few very minor UI issues

____________________________________________________________________________________________________________

- [ ] Line height on the chevron icon 😉  Nothing major!
screen shot 2018-06-20 at 2 03 41 pm

____________________________________________________________________________________________________________

- [ ] Adding hover states with reduced opacities for the social icons would be a neat touch 😄 

____________________________________________________________________________________________________________

@pelsasser we can definitley iterate on the sidebar 😃  
Also, do you think the placement of the categoies section in mobile resolution is right? Its placed after the first post currently. We could also do it by placing in the end of blog cards before the 'Load more' Button. It works either way!

**Comment by hbeckeri** [comment]
@nitinrgupta I've addressed your most recent PR concerns

**Comment by eswarasai** [comment]
@nitinrgupta -- Anything else you want to add here or is it good to be merged?

@pelsasser -- Feel free to merge this once we get it approved from Nitin. We might have to uncomment the `/blog` route in `static.config.json` as well as update the link on the Navbar to `/blog` instead of medium.com.

**Comment by nitinrgupta** [comment]
@hbeckeri any thoughts on utilizing the `styled components` more effectively? By using them, generally CSS classnames and ids are avoided in the markup.

@eswarasai in the interest of getting things moving should we handle that some other time? @pelsasser you can take a call on this. 

Apart from that, everything LGTM 👍

**Comment by eswarasai** [comment]
@hbeckeri -- How much efforts do you think it might take for you to update the code to use `styled-components`? I'd recommend having similar pattern across the codebase but I'd wait for @pelsasser to decide on the priority based on your response of the efforts. Thanks!

**Comment by hbeckeri** [comment]
@eswarasai @eswarasai I've seen some inconsistency with styling all over the codebase. For example the team page uses inline styles compared to other that use styled-components and i've seen others that use classNames and ids. I've coded the styles two times now and both times I've tried to follow patterns that already existed. Im happy to rewrite them again, however I'd like you to choose a reference file for me to follow the pattern of.

**Pull request merged**